### DISCUSSION TITLE
how to use hooks to get nested fields

### DISCUSSION THREAD

**Discussion by LimaneGaya** [pocketbase/pocketbase#4398] [Q&A] [answered]
i have a database that have users, conversations and messages. 

i want to set up hooks that sent http requests every time a message, i tried using onRecordAfterCreateRequest like this :

```
onRecordAfterCreateRequest((e) => {
    const messageId = e.record.publicExport()["conversation"]
    const record = $app.dao().findRecordById("converstion", messageId)
    const participants = record.publicExport()["participants"]
    for (const person in participants) {
        const user = $app.dao().findRecordById("users", person)
        console.log(user.publicExport.toString())
    }
    

}, "messages")
```
but it doesn't work, is there a better way to do that, i'm not familiar with either javascript or go and i find this really hard to debug.
how can i get conversation record from the new message record created then get users that are inside the conversation.

i'd really appreciate any help, thank you

**Comment by ganigeorgiev** [comment, marked as answer]
> but it doesn't work

What is the error message that is returned? Did you check the _Admin UI > Logs_ section?

> i find this really hard to debug

Start the executable with `--dev` to print the running SQL queries and logs in the terminal.

In any case `person` in your example will be the index of the `participants` array item, not the id.
Replace `in` with `of` in the loop and the user should return the related record.

Also make sure that there is no typo in your collection names (inspect the returned error message).

Additionally, note that you don't have to export all fields just to take a single value. Calling `record.get("yourField")` (or the other [get helpers](https://pocketbase.io/docs/js-records/#getset-record-fields)) should be sufficient.

Below is a slightly modified version of your code:

```js
onRecordAfterCreateRequest((e) => {
    const messageId = e.record.get("conversation");
    
    // note: make sure that the collection name is correct
    const conversation = $app.dao().findRecordById("converstion", messageId)
    
    const participants = conversation.get("participants")
    // or replace directly with:
    // const users = $app.dao().findRecordsByIds("users", participants)
    for (const person of participants) {
        const user = $app.dao().findRecordById("users", person)
        console.log(user)
    }
}, "messages")
```

If the above doesn't help, please elaborate what is the returned error and export your collections configuration as json (Admin UI > Settings > Export collections).

**Reply by LimaneGaya** [reply]
It solved now 
> In any case person in your example will be the index of the participants array item, not the id.
> Replace in with of in the loop and the user should return the related record.

this was what i was doing wrong, and the suggestion to edit the code is helpful. Thanks for the answer.
